subprojects/docs/src/docs/userguide/managing_transitive_dependencies.adoc

```diff
@@ -165,7 +165,66 @@ In the example, the versions of `gson` and `dom4j` are provided by the Spring Bo
 
+However often BOMs are not only providing versions as recommendations, but also a way to override any over version found in the graph. You can enable this behavior by using the `enforcedPlatform` keyword, instead of `platform`, when importing the BOM:
+
+.Importing a BOM, making sure the versions it defines override any other version found in the graph
+====
+include::sample[dir="userguide/dependencyManagement/managingTransitiveDependencies/constraintsFromBOM/groovy",files="build.gradle[tags=dependency-on-bom]"]
+include::sample[dir="userguide/dependencyManagement/managingTransitiveDependencies/constraintsFromBOM/kotlin",files="build.gradle.kts[tags=dependency-on-bom]"]
+====
+
+[[sec:version_alignment]]
+== Dependency version alignment
+=== Handling inconsistent module versions
+Gradle supports aligning versions of modules which belong to the same "platform". It is often preferable, for example, that the API and implementation modules of a component are using the same version. However, because of the game of transitive dependency resolution, it is possible that different modules belonging to the same platform end up using different versions.
+For example, your project may depend on the `jackson-databind` and `vert.x` libraries, as illustrated below:
+
+.A dependency on the Jackson library
+====
+include::sample[dir="userguide/dependencyManagement/managingTransitiveDependencies/dependencyAlignment/groovy",files="build.gradle[tags=dependencies]"]
+include::sample[dir="userguide/dependencyManagement/managingTransitiveDependencies/dependencyAlignment/kotlin",files="build.gradle.kts[tags=dependencies]"]
+====
+
+Because `vert.x` depends on `jackson-core`, we would actually resolve the following dependency versions:
+
+- `jackson-core` version `2.9.5` (brought by `vertx-core`)
+- `jackson-databind` version `2.9.5` (by conflict resolution)
+- `jackson-annotation` version `2.9.0` (dependency of `jackson-databind:2.9.5`)
+
+It's easy to end up with a set of versions which do not work well together. To fix this, Gradle supports dependency version alignment, which is supported by the concept of platform. A platform represents a set of modules which "work well together". Either because they are actually published as a whole (when one of the members of the platform is published, all other modules are also published with the same version), or because someone tested modules and indicates that they work well together (typically, the Spring Platform).
+
+=== Declaring participation in a platform
+
+We can fix the example above by declaring that all Jackson modules "belong to" the same platform.
+
+.An alignment rule
```

Comment:
I think we probably ought to always say "dependency alignment" rather than just "alignment" because that word probably has a bunch of other meanings to the reader.